jax/_src/api.py

```diff
@@ -151,7 +153,7 @@ def jit(
   backend: str | None = None,
   inline: bool = False,
   abstracted_axes: Any | None = None,
-) -> pjit.JitWrapped:
+) -> pjit.JitWrapped[_P, _OutT]:
```

Comment:
I'm seeing an error here in pytype that's blocking any further testing:
```
/jax/_src/api.py:156: error: in <module>: class JitWrapped is not indexable [not-indexable]
  ('JitWrapped' does not subclass Generic)
```

Reply:
`JitWrapped` does in fact inherit from `Generic`, so I suspect there's something deeper going on that's preventing `pytype` from seeing that within the package structure of bazel-based builds.

Reply:
One potential solution (although ugly) could be to make `Wrapped` inherit directly from `Generic`, in addition to `Protocol` (which itself inherits from `Generic`). 
Should we test this out?

Reply:
@superbobry do you have thoughts here?

One thing we could do is just ignore this error in PyType, and then within pytype `jit` would essentially be treated as `Any` until we fix the bazel build dependencies. That would let us have typed JIT externally without having to fix all the internal errors it would uncover.

Reply:
I think inheriting from `Generic` and listing type parameters makes sense. This is always fine to do for a generic class.

Note sure about inheriting from `Protocol`, because we don't want `JitWrapped` to be a protocol itself (it does implement the `stages.Wrapped` protocol, though).

Reply:
If that doesn't work, we can try raising this with the pytype team.

Reply:
Ok, I'll give this a shot. Thanks!

Reply:
I pushed a new version of this with `JitWrapped` inheriting from Generic directly (in addition to the existing `JitWrapped` -> `Wrapped` -> `Protocol` -> `Generic` chain):  5d9f7629f2b3154c2de961197e1347927deee02d

Reply:
Oh and btw @superbobry, I meant to say that JitWrapped already subclasses Generic through Protocol, not that we should make it subclass both Generic and Protocol, if that makes sense.

Reply:
I think @superbobry was saying that `JitWrapped` should **not** inherit from `Wrapped`, because `Wrapped` is a protocol.

Reply:
I didn't say that, but I agree with @jakevdp in principle :)